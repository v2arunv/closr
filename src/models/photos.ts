export interface IPhoto {
    albumId: number,
    id: number,
    title: string,
    url: string,
    thumbnailUrl: string,
}

export interface IAlbum {
    userId: number,
    id: number,
    title: string,
}

export interface IPhotoAlbum {
    album: IAlbum,
    photos: Array<IPhoto>
}

export const convertAlbumPayloadIntoModel = (payload: any) => {
    const {
        userId,
        id,
        title
    } = payload;
    return <IAlbum> {
        userId,
        id,
        title,
    }
};

export const convertPhotoPayloadIntoModel = (payload: any) => {
    const {
        albumId,
        id,
        title,
        url,
        thumbnailUrl,
    } = payload;
    return <IPhoto> {
        albumId,
        id,
        title,
        url,
        thumbnailUrl,
    }
};