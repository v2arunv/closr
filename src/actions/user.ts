import {IUser} from "../models/users";

export interface IUserAction {
    type: string,
    user: IUser,
}

export const setUser = (user: IUser): IUserAction => {
    return <IUserAction> {
        type: 'SET_USER',
        user: user,
    }
};