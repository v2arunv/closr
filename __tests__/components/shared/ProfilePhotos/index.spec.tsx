import React from 'react';
import renderer from 'react-test-renderer';
import user from '@mocks/users';
import { ProfilePhotos }from "@sharedComponents/ProfilePhotos";
import {IAlbum, IPhoto, IPhotoAlbum} from "@models/photos";
import { allMocks } from "@mocks/photos";

const album: IAlbum = {
    userId: 1,
    id: 1,
    title: 'Mock Album 1',
};

const photos: Array<IPhoto> = [ allMocks[0], allMocks[1]];

const photoAlbum: IPhotoAlbum = {
    album,
    photos,
};


const props = {
    user,
    getUserPhotos: () => {},
    isError: false,
    isLoading: false,
    albums: [photoAlbum],
    onPress: () => {},
};

test('renders correctly', () => {
    const tree = renderer.create(
        <ProfilePhotos
            {...props}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});