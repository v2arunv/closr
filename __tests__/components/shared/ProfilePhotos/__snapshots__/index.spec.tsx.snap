// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "marginTop": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "borderBottomColor": "#ddd",
        "borderBottomWidth": 0,
        "elevation": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
        "marginTop": 5,
      }
    }
  >
    Photos
  </Text>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text>
        Mock Album 1
      </Text>
      <RCTScrollView
        horizontal={true}
        style={
          Object {
            "height": 70,
          }
        }
      >
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 60,
                "margin": 5,
                "opacity": 1,
                "width": 60,
              }
            }
          >
            <Image
              defaultSource={
                Object {
                  "testUri": "../../../src/assets/images/beach-parasol-water-1.png",
                }
              }
              source={
                Object {
                  "uri": "https://via.placeholder.com/150/92c952",
                }
              }
              style={
                Object {
                  "height": 60,
                  "width": 60,
                }
              }
            />
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 60,
                "margin": 5,
                "opacity": 1,
                "width": 60,
              }
            }
          >
            <Image
              defaultSource={
                Object {
                  "testUri": "../../../src/assets/images/beach-parasol-water-1.png",
                }
              }
              source={
                Object {
                  "uri": "https://via.placeholder.com/150/771796",
                }
              }
              style={
                Object {
                  "height": 60,
                  "width": 60,
                }
              }
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
